### PR TITLE
Feat/add giverole command

### LINKED ISSUE #1006: Give Role Command
`s!giverole <user> <role name>`

This command should check if the user has permission to give roles and is able to give that role to that user (role can't be above the user's highest role, except if they're admin. Target user can't be above the user either)

### PR DESCRIPTION
**Pull request by estevam31**
Closes #1006

### CODE REVIEW

File: src/commands/moderation/giverole.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        type: 'member', acceptBot: true, missingError: 'commands:giverole.missingUser'
+      }, {
+        type: 'role', full: true, missingError: 'errors:invalidRole'
+      }]
+    })
+  }
+
+  async run ({ channel, member, author, t }, targetMember, role) {
+    const embed = new SwitchbladeEmbed(author)
+    await targetMember.addRole(role).then(modifiedMember => {
```
Comment (davipatury) [inline src/commands/moderation/giverole.js]: Whenever you use async/await, you don't have to use `.then` or `.catch` you can use the simple try/catch. Ex.:
```js
try {
  await targetMember.addRole(role)
  // do other things
} catch (e) {
  // handle error
}
```

File: src/commands/moderation/giverole.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      if (!member.hasPermission('ADMINISTRATOR', false, true, false)) {
+        if (member.highestRole.comparePositionTo(role) <= 0) throw new CommandError(t('commands:giverole.roleAboveAuthors'))
+        if (member.highestRole.comparePositionTo(member.highestRole) <= 0) throw new CommandError(t('commands:giverole.roleBelow'))
+      }
```
Comment (davipatury) [inline src/commands/moderation/giverole.js]: This verification process should be made before adding the role to the target member.

And the condition in line 22 is non-sense, you are comparing `member.highestRole` to itself: **`member.highestRole`**`.comparePositionTo(`**`member.highestRole`**`)`

Comment (estevam31) [reply src/commands/moderation/giverole.js]: Oh shit, I just changed the variable names and forgot this one lol.

File: src/commands/moderation/giverole.js (hunk 1 above)
Comment (estevam31) [reply src/commands/moderation/giverole.js]: Just sent this change

**Comment by pedrofracassi** [comment]
So, how's this going?

**Comment by metehus** [comment]
@davipatury re-review?

**Pull request closed without merging**